cg/cli/set/case.py

```diff
@@ -77,18 +80,25 @@ def set_case(
     if panel_abbreviations:
         update_panels(case=case, panel_abbreviations=panel_abbreviations, status_db=status_db)
 
-    if priority:
+    if isinstance(priority, Priority):
         update_priority(case=case, priority=priority)
 
     status_db.session.commit()
 
 
-def abort_on_empty_options(options: list[str]) -> None:
-    if not any(options):
+def abort_on_empty_options(
+    options: list[str], bypass_check: Callable[[any], bool], priority: Priority | None
```

Comment:
I get that it is expandable but the bypass_check parameter is quite hard to understand. Why are we allowing for this to be passed as a parameter rather than always just checking if the priority is research?